internal/runtime/internal/controller/loader.go

```diff
@@ -260,7 +262,13 @@ func (l *Loader) Apply(options ApplyOptions) diag.Diagnostics {
 	l.componentNodes = components
 	l.serviceNodes = services
 	l.graph = &newGraph
-	l.cache.SyncIDs(componentIDs)
+	err := l.cache.SyncIDs(componentIDs)
+	if err != nil {
```

Comment:
Should this return immediately?

Reply:
I doesnt have to because it's adding to the diags so Alloy will exit anyway but I agree that's a little bit cleaner so I will make the change